lib/posix/posix.nim

```diff
@@ -1110,5 +1110,8 @@ proc getrlimit*(resource: cint, rlp: var RLimit): cint
       {.importc: "getrlimit",header: "<sys/resource.h>".}
   ## The getrlimit() system call gets resource limits.
 
+when defined(linux) or defined(bsd):
+  proc sethostname*(name: cstring, len: csize_t): cint {.importc, header: "<unistd.h>", sideEffect.}
```

Comment:
* available on osx but with a different type for namelen:
```
     int
     gethostname(char *name, size_t namelen);

     int
     sethostname(const char *name, int namelen);
```

> HISTORY The gethostname() function appeared in 4.2BSD.  The namelen argument to gethostname() was changed to size_t in FreeBSD 5.2 for alignment with IEEE Std 1003.1-2001 (``POSIX.1'').


* on linux:
   int gethostname(char *name, size_t len);
   int sethostname(const char *name, size_t len);

- [ ] (pre-existing)
```
posix.nim:496:6:proc gethostname*(a1: cstring, a2: int): cint {.importc, header: "<unistd.h>", sideEffect.}
```
is wrong (int vs csize_t)


## proposal
since `namelen` has different types depending on osx vs linux, how about adding a wrapper:
```nim
proc setHostName*(name: string): cint
```
which will call into those depending on linux vs other

Reply:
Thanks for your proposal; I've decided to do it even more high-level (raise an exception instead of returning `cint`). See the code.